test/k8sT/microscope.go

```diff
@@ -56,14 +53,28 @@ var _ = Describe(microscopeTestName, func() {
 		microscopeErr, microscopeCancel := kubectl.MicroscopeStart()
 		Expect(microscopeErr).To(BeNil(), "Microscope cannot be started")
 
-		time.Sleep(time.Second * 10)
+		err := helpers.WithTimeout(func() bool {
+			res := kubectl.ExecPodCmd("kube-system", "microscope", "pgrep -f microscope")
+			return res.WasSuccessful()
+		}, "failed to start microscope",
+			&helpers.TimeoutConfig{
+				Ticker:  5,
+				Timeout: 120,
+			})
+
+		Expect(err).To(BeNil())
 
 		kubectl.ValidateNoErrorsOnLogs(CurrentGinkgoTestDescription().Duration)
 		Expect(microscopeCancel()).To(BeNil(), "cannot stop microscope")
 
-		time.Sleep(time.Second * 10)
-
-		res := kubectl.ExecPodCmd("kube-system", "microscope", "pgrep -f microscope")
-		res.ExpectFail()
+		err = helpers.WithTimeout(func() bool {
+			res := kubectl.ExecPodCmd("kube-system", "microscope", "pgrep -f microscope")
+			return !res.WasSuccessful()
+		}, "failed to close microscope",
```

Comment:
Similar to above: error message is somewhat cryptic in relation to what this is checking for. I'd recommend something like "found running microscope processes; no microscope processes should be running" ( ? )